core/src/interest.rs

```diff
@@ -68,6 +76,25 @@ impl Interest {
     }
 }
 
+impl std::fmt::Debug for Interest {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if f.alternate() {
+            f.debug_struct("Interest")
```

Comment:
So instead of having successors for the individual bits just have a decode method that returns a new struct that is all the fields? 

I could do that, for now only the `range` field is decoded so that is why I left as individual accessor methods.